reqMgrClient.py

```diff
@@ -902,73 +932,97 @@ def assignWorkflow(url, workflowname, team, parameters ):
+        else:
+            params = {"requestName" : workflowname,"status" : "closed-out"}
+            data = requestManager1Put(url,"/reqmgr/reqMgr/request", params)
     return data
 
 def closeOutWorkflowCascade(url, workflowname):
-    """
-    Closes out a workflow, it will search for any Resubmission requests 
-    for which the given request is a parent and announce them too.
-    """
-    params = {"requestName" : workflowname, "cascade" : True}
-    data = requestManagerPost(url,"/reqmgr/reqMgr/closeout", params)
-    return data
+    return closeOutWorkflow(url, workflowname, True)
 
-def announceWorkflow(url, workflowname):
+def announceWorkflow(url, workflowname, cascade=False):
     """
     Sets a workflow state to announced
     This does not care about cascade workflows
     """
-    params = {"requestName" : workflowname,"status" : "announced"}
-    data = requestManagerPut(url,"/reqmgr/reqMgr/request", params)
+    if isRequestMgr2Request(url, workflowname): 
+        params = {"RequestStatus" : "announced",
+                  "cascade": cascade}
+        data = requestManagerPut(url,"/reqmgr2/data/request/%s"%workflowname, params)
+    else:
+        if cascade:
+            params = {"requestName" : workflowname,"cascade" : cascade}
+            data = requestManager1Post(url,"/reqmgr/reqMgr/announce", params)
+        else:
+            params = {"requestName" : workflowname,"status" : "announced"}
+            data = requestManager1Put(url,"/reqmgr/reqMgr/request", params)
     return data
 
 def announceWorkflowCascade(url, workflowname):
-    """
-    Sets a workflow state to announced, it will search for any Resubmission requests 
-    for which the given request is a parent and announce them too.
-    """
-    params = {"requestName" : workflowname, "cascade" : True}
-    data = requestManagerPost(url,"/reqmgr/reqMgr/announce", params)
-    return data
+    return announceWorkflow(url, workflowname,True)
 
 
 def setWorkflowApproved(url, workflowname):
     """
     Sets a workflow state to assignment-approved
     """
-    params = {"requestName" : workflowname,"status" : "assignment-approved"}
-    data = requestManagerPut(url,"/reqmgr/reqMgr/request", params)
+    if isRequestMgr2Request(url, workflowname):
+        params = {"RequestStatus" : "assignment-approved"}
+        data = requestManagerPut(url,"/reqmgr2/data/request/%s"%workflowname, params)
+    else:
+        params = {"requestName" : workflowname,"status" : "assignment-approved"}
+        data = requestManager1Put(url,"/reqmgr/reqMgr/request", params)
     return data
 
 def setWorkflowForceComplete(url, workflowname):
-    """
-    Sets a workflow state to force-complet
-    """
-    params = {"requestName" : workflowname,"status" : "force-complete"}
-    data = requestManagerPut(url,"/reqmgr/reqMgr/request", params)
-    return data
+    return forceCompleteWorkflow(url, workflowname)
 
 def setWorkflowRunning(url, workflowname):
```

Comment:
Such method should not exist, this transition is made by the agent, not by human.
BTW, "running" state has been deprecated long ago, it's now either "running-open" or "running-closed"